### PR TITLE
Do not verify the signature in SignedLogRoot.

### PR DESCRIPTION
**Pull request by pphaneuf**
Part of #2450.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Hmm, not sure about this one. What's the bigger plan?

Was the signature verification also sort of checking the "safety" of the TLS-encoded blob? I think it came in pair with the TLS encoding. In this regard, maybe the TLS encoding is also not needed on its own?

**Comment by pphaneuf** [comment]
The TLS encoding was needed for stability of the signature (but all along, since it was being kept in a `bytes` field, it could also have been a serialised protobuf). You are correct that without signature validation, TLS encoding is not necessary, but I'm going one step at a time... 😄

**Comment by pphaneuf** [comment]
FYI, the plan is the same as usual, removing code "from the front": remove the client code checking the signatures, remove the signature fields in the RPCs, remove the infrastructure supporting filling out those fields, etc...

File: crypto/signer_test.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func TestSignLogRoot(t *testing.T) {
 			t.Errorf("len(sig): %v, want > 0", got)
 		}
 		// Check that the signature is correct
-		if _, err := VerifySignedLogRoot(key.Public(), crypto.SHA256, slr); err != nil {
+		if err := Verify(key.Public(), crypto.SHA256, slr.LogRoot, slr.LogRootSignature); err != nil {
```
Comment (pavelkalinnikov) [inline crypto/signer_test.go:110]: Is this signature/key also to be removed (separately)?

**Review by pavelkalinnikov — APPROVED** [review]

File: crypto/signer_test.go (hunk 1 above)
Comment (pphaneuf) [reply crypto/signer_test.go:110]: Yes, I wanted to just remove the signature verification in the PR, but without affecting the signing itself (including the code testing the signing).

Later PRs will remove the signing itself, as well as the key management infrastructure (which is unfortunately quite plentiful and deeply rooted!).

**Pull request merged**